### ISSUE TITLE
Cannot find ZSTD_maxCLevel() function

### ISSUE THREAD

**Issue by FrancescAlted** [Cyan4973/zstd#254]
I am trying to use the supposedly public `ZSTD_maxCLevel` (defined  in zstd.h), but I am getting this error:

```
/home/faltet/blosc/c-blosc/blosc/blosc.c: In function ‘zstd_wrap_compress’:
/home/faltet/blosc/c-blosc/blosc/blosc.c:512:44: warning: implicit declaration of function ‘ZSTD_maxCLevel’ [-Wimplicit-function-declaration]
   clevel = (clevel < 9) ? clevel * 2 - 1 : ZSTD_maxCLevel();
```

I am not sure what's going on here.  I would say that my code is correct:

```
static int zstd_wrap_compress(const char* input, size_t input_length,
                              char* output, size_t maxout, int clevel) {
  size_t code;
  clevel = (clevel < 9) ? clevel * 2 - 1 : ZSTD_maxCLevel();
  code = ZSTD_compress(
      (void*)output, maxout, (void*)input, input_length, clevel);
  if (ZSTD_isError(code)) {
    fprintf(stderr, "error compressing with Zstd: %s \n", ZSTD_getErrorName(code));
    return 0;
  }
  return (int)code;
}
```

but maybe I am missing something?

**Comment by Cyan4973** [comment]
I guess the issue is that `ZSTD_maxCLevel()` is part of advanced / experimental API.

If you just `#include "zstd.h"`, only the Stable API is included.

To access experimental API, it's necessary to `#define ZSTD_STATIC_LINKING_ONLY` before `#include` statement.

This is a protection, to ensure only stable API will be used with a dynamic library.

**Comment by FrancescAlted** [comment]
Ah, makes sense.  Now that we are talking about this, I think it would make sense to move `ZSTD_MAX_CLEVEL` to zstd.h (to the stable API I mean).  Or is there any reason you don't want to make this move?

**Comment by Cyan4973** [comment]
This kind of discussion is exactly what's needed to move forward items from "experimental" to stable.

I was unsure if the concept of checking for `ZSTD_maxCLevel()` would remain valid, or if it would converge towards something more static / stable (like the classical 1 to 9).
It's still not completely clear, but I feel ``ZSTD_maxCLevel()` remains compatible with any scenario, so it seems a good candidate for "stable".

The prototype looks simple enough.
The only remaining question is about the return type : currently it's an `unsigned`, but compression level is typed as an `int`, in order to allow _negative compression levels_ in the future (similar to what lz4 provides with _acceleration_ values). So, maybe `ZSTD_maxCLevel()` should be typed as an `int` too, this way its result can be directly fed to `ZSTD_compress()` without any risk of sign-conversion warning.

**Comment by Cyan4973** [comment]
`ZSTD_maxCLevel()` has been promoted to "stable" API.
It will be present there starting from v0.8.0.

**Comment by Cyan4973** [comment]
Released within v0.8.0
